packages/react-components/src/components/SendBox.tsx

```diff
@@ -233,12 +233,20 @@ export const SendBox = (props: SendBoxProps): JSX.Element => {
     }
 
     // we dont want to send empty messages including spaces, newlines, tabs
-    if (!EMPTY_MESSAGE_REGEX.test(textValue)) {
-      onSendMessage && onSendMessage(textValue);
+    // Message can be empty if there is a valid file upload
+    if (!EMPTY_MESSAGE_REGEX.test(textValue) || hasFile()) {
+      // Chat SDK doesn't send messages with empty content. For sending files without text message,
+      // we need to send a message with a space.
+      if (EMPTY_MESSAGE_REGEX.test(textValue)) {
+        onSendMessage && onSendMessage(' ');
```

Comment:
quick question: should we send metadata from here(onSendMessage) with text all together? It looks more readable if user want to implement a file upload from component level

Reply:
Thought about it.
Metadata is not really a part of the message or sendbox from a pure UI perspective.
It is an underlying implementation detail that contoso needn't care about.

Also, if contoso were to use this component without composite and implement file sharing, they would again not need metadata here. :)

Reply:
I am thinking of a scenario where contoso wants to implement this with component + stateful client (still use our acs), I believe the recommended from us is still using metadata right? I am thinking if we could provide an optional metadata prop in sendBox component, then sendMessage will be sending that metadata if user provides any
But yea, this is not related to this PR, let's discuss the topic when we are writing a fully functional component-level uploading sample for our contoso

Reply:
That's a great point @PorterNan 
Can I create a fast follow up PR for this?